### DISCUSSION TITLE
Is it possible to totally clear scripting engine?

### DISCUSSION THREAD

**Discussion by NotSoRandomOne** [Stellarium/stellarium#3350] [General]
When scripting, it appears that Stellarium 'holds on' to previous code. For instance, given this simple script:

<pre>main();

function main() {
   test();
   }

function test() {
   core.debug(core.getDate());
   }</pre>

if you run it once, then comment out the 'test' function, and execute it again, it will execute the core.debug line even though it is now commented out. There seems to be another issue having something to do with if you create a variable at a local function scope then later move that to a global scope, it is not recognized as a global variable for some reason.  I don't fully understand that issue (which drove me crazy for a couple days), but maybe totally clearing out the scripting engine would put it in a blank slate so those problems go away.

Thanks,
David

**Comment by gzotti** [comment]
Depending on version, Stellarium may either use Qt5's QtScript or Qt6's Qml modules. 
Most scripts don't use additional functions, so this issue may not have surfaced for us in the past several years. 
We 3 are no JS experts. Feel invited to explore and improve the Scripting part of Stellarium.

**Reply by NotSoRandomOne** [reply]
Thank you. I will try, but doubt I'll have time. My guess is it isn't anything to do with Stellarium itself, but something to do with the interactions with Qt. If there is a simple way to reset the Qt script engine between script runs it might fix the issues.  Closing and reopening Stellarium seems to fix the issue with the example script up above.

**Reply by gzotti** [reply]
"I have no time": well, me too, but also "not my problem"... This project lives from "I will have a look into this" by the same people who face such problems, and then steal their own time to fix them.  We have no deeper insight into Qt's inner workings than what is available online.
